cicd/base/app/base.yaml

```diff
@@ -0,0 +1,357 @@
+AWSTemplateFormatVersion: '2010-09-09'
+
+Description: Base resources for the Userdetails service
+
+Parameters:
+  pBuild:
+    Type: String
+    Description: The build number
+  pCleanBranch:
+    Type: String
+    Description: The clean branch, can be used in resource names
+  pClusterOidcIssuerUrl:
+    Type: String
+    Description: The OIDC issuer URL for the cluster
+  pEnvironment:
+    Type: String
+    Description: The AWS environment this belongs to
+  pProductName:
+    Type: String
+    Description: The name of the product
+  pOidcConnectProvider:
+    Type: String
+    Description: The OIDC connect provider
+  pServiceAccountName:
+    Type: String
+    Description: The name of the service account
+  pCognitoStackName:
+    Type: String
+    Description: The name of the Cognito stack
+  pAppClientDomain:
+    Type: String
+    Description: The domain for the app client
+  pSecretName:
+    Type: String
+    Description: The name of the secret
+  pDefaultLogoutUrl:
+    Type: String
+    Description: The name of the default logout url
+
+Conditions:
+
+  IsDev: !Equals
+    - !Ref pEnvironment
+    - development
+  NotProd: !Not
+    - !Equals
+      - !Ref pEnvironment
+      - production
+  IsFirstBuild: !Equals [!Ref pBuild, "1"]
+
+Resources:
+
+  UserdetailsRepository:
+    Type: AWS::ECR::Repository
+    Properties:
+      EmptyOnDelete: !If [ NotProd, true, false ]
+      EncryptionConfiguration:
+        EncryptionType: AES256
+      ImageScanningConfiguration:
+        ScanOnPush: true
+      RepositoryName: !Sub
+                         - ${pProductName}-${ResourceName}
+                         - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
+      RepositoryPolicyText:
+        Version: 2012-10-17
+        Statement:
+          - Sid: AllowPublicPull
+            Effect: Allow
+            Principal: '*'
+            Action:
+              - 'ecr:GetDownloadUrlForLayer'
+              - 'ecr:BatchGetImage'
+              - 'ecr:BatchCheckLayerAvailability'
+      Tags:
+        - Key: Environment
+          Value: !Ref pEnvironment
+        - Key: Branch
+          Value: !Ref pCleanBranch
+
+  UserdetailsSecret:
+    Type: 'AWS::SecretsManager::Secret'
+    Properties:
+      Name: !Sub ${pSecretName}
+      Description: !Sub Userdetails app ${pEnvironment} secrets
+      SecretString: !If
+              - IsFirstBuild
+              - |
+                {
+                   "userdetails_oidc_secret" : "",
+                   "userdetails_webservice_secret": "",
+                   "security_admin_password" : "",
+                   "flickr_secret" : "",
+                   "inaturalist_secret" : "",
+                   "recaptcha_secret": ""
+                }
+              - !Ref "AWS::NoValue"
+
+  UserdetailsPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: !Sub Permissions to deploy ${pProductName}
```

Comment:
This policy looks like it's for userdetails execution rather than deploy

Reply:
Fixed.